pkg/controller/beat/common/stackmon/stackmon_test.go

```diff
@@ -0,0 +1,221 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+package stackmon
+
+import (
+	"context"
+	_ "embed"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+
+	"github.com/elastic/cloud-on-k8s/v2/pkg/apis/beat/v1beta1"
+	commonv1 "github.com/elastic/cloud-on-k8s/v2/pkg/apis/common/v1"
+	v1 "github.com/elastic/cloud-on-k8s/v2/pkg/apis/common/v1"
+	esv1 "github.com/elastic/cloud-on-k8s/v2/pkg/apis/elasticsearch/v1"
+	"github.com/elastic/cloud-on-k8s/v2/pkg/controller/common/stackmon"
+	"github.com/elastic/cloud-on-k8s/v2/pkg/controller/elasticsearch/bootstrap"
+	"github.com/elastic/cloud-on-k8s/v2/pkg/utils/k8s"
+)
+
+func TestMetricBeat(t *testing.T) {
+	containerFixture := corev1.Container{
+		Name:  "metricbeat",
+		Image: "docker.elastic.co/beats/metricbeat:8.2.3",
+		Args:  []string{"-c", "/etc/metricbeat-config/metricbeat.yml", "-e"},
+		Env: []corev1.EnvVar{
+			{
+				Name: "POD_IP",
+				ValueFrom: &corev1.EnvVarSource{
+					FieldRef: &corev1.ObjectFieldSelector{
+						APIVersion: "v1",
+						FieldPath:  "status.podIP",
+					},
+				},
+			},
+			{
+				Name: "POD_NAME",
+				ValueFrom: &corev1.EnvVarSource{
+					FieldRef: &corev1.ObjectFieldSelector{
+						APIVersion: "v1",
+						FieldPath:  "metadata.name",
+					},
+				},
+			},
+			{
+				Name: "NODE_NAME",
+				ValueFrom: &corev1.EnvVarSource{
+					FieldRef: &corev1.ObjectFieldSelector{
+						APIVersion: "v1",
+						FieldPath:  "spec.nodeName",
+					},
+				},
+			},
+			{
+				Name: "NAMESPACE",
+				ValueFrom: &corev1.EnvVarSource{
+					FieldRef: &corev1.ObjectFieldSelector{
+						APIVersion: "v1",
+						FieldPath:  "metadata.namespace",
+					},
+				},
+			},
+		},
+		VolumeMounts: []corev1.VolumeMount{
+			{
+				Name:      "beat-metricbeat-config",
+				ReadOnly:  true,
+				MountPath: "/etc/metricbeat-config",
+			},
+		},
+	}
+	beatYml := `metricbeat:
+  modules:
+  - hosts:
+    - http://localhost:5066
+    metricsets:
+    - stats
+    - state
+    module: beat
+    period: 10s
+    xpack:
+      enabled: true
+monitoring:
+  cluster_uuid: abcd1234
+output:
+  elasticsearch:
+    hosts:
+    - es-metrics-monitoring-url
+    password: es-password
+    username: es-user
+`
+	beatSidecarFixture := stackmon.BeatSidecar{
+		Container: containerFixture,
+		ConfigSecret: corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "beat-beat-monitoring-metricbeat-config",
+				Namespace: "test",
+				Labels: map[string]string{
+					"common.k8s.elastic.co/type":                "elasticsearch",
```

Comment:
These labels seem wrong. Why is this of  type "elasticsearch". I understand this not something you introduced in this PR but this needs at least a follow up issue to clean this up.

Reply:
https://github.com/elastic/cloud-on-k8s/issues/5967